andela-wmaina/document-management
# Develop

#### What does this PR do?
- Adds mocha tests for user, document and role routes
- Adds user and document pagination routes
- Adds user and document search routes
- Eslints all files

 #### Description of Task to be completed?
- Create test suites for all routes
- Implement the following routes:
   - GET /users/?limit={integer}&offset={integer}
   - GET /documents/?limit={integer}&offset={integer}
   - GET /search/users/?q={username}
   - GET /search/documents/?q={doctitle}
- Include an eslint file that extends the airbnb standard in the root directory to lint server side code

 #### How should this be manually tested?
- Clone the repo
- run `npm install`
- run `npm run start:dev`
- to run tests
   - run `npm test`
- to run pagination routes:
   - Copy/Paste the above routes and replace the {integer} with a number where:
      limit: the number of records you want to query 
      offset: How many instances should be skipped
- to run search routes:
    - for users: replace {username} with the name of the user you want to search
    - for documents: replace {title} with the title of the doc you are searching for

 #### What are the relevant pivotal tracker stories?

- 140472095
- 141777673
- 141880597
- 141881983
